kanidm_unix_int/nss_kanidm/src/lib.rs

```diff
@@ -10,22 +10,34 @@
 #![deny(clippy::needless_pass_by_value)]
 #![deny(clippy::trivially_copy_pass_by_ref)]
 
+#[cfg(target_family = "unix")]
```

Comment:
Feels a bit rough to have so many cfg statements here, do we actually need to do something silly like break this into modules that are cfg gated? 

Something like:

```
#[cfg(target_family = "unix")]
#[macro_use]
extern crate ...

#[cfg(target_family) ]
mod blahblah {
     // all the use statements, code, hooks etc.
}

#[cfg(not(target_family)) ]
mod blahblah
{
     do nothing? 
}
```

Reply:
yeah, it's a big clusterf*ck and was me just trying to yolo the whole thing on those modules - I'm poking around to find better ways of doing it in general.

Reply:
> Feels a bit rough to have so many cfg statements here, do we actually need to do something silly like break this into modules that are cfg gated?

That is generally how it is done, yes. You build a module that defines a shared common abstraction and then you have each platform provide the relevant implementations to back up this abstraction, so you only need to write something like `#[cfg] use platform::imp as module;`.

Reply:
@workingjubilee The problem here is there is no common abstraction for pam or nsswitch on windows, which is why we raised the conditional compilation target issue. We want to completely skip these workspace members on windows rather than try to cfg define them out and make an empty .dll. So in this case the cfg around the high level module break out is still a bit of a "hack" for us :(

Reply:
Oof. Yeah, that's fair. Sometimes doing things with computers is Like This.

Reply:
Absolutely. Especially when we are writing something that is very platform dependent and targeted, we get bitten by things like this. In the opposite, we are starting to see interest in development of a windows specific authentication module, which means we'll have the reverse issue of a target that's windows only.

Thus, why we opened the rust (now cargo) issue about platform specific build targets for workspace members, since it would be the cleanest option for us, and quite likely, many others :)

Reply:
Cleaned this up a bunch. :D